### PR TITLE
Redesign Region API and add regioned objects

### PR DESCRIPTION
**Pull request by thedadams**
The Region API is overhauled here such that there is no longer any RegionInstances. Additionally, a user will not be able to create a region from the CLI, and a hard-coded "local" region will exist on installation.

Finally, the various objects that must be specific to a region implement an interface for detection purposes.

### Checklist
- [ ] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in paranthesis, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [ ] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keyworkds](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [ ] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [ ] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/apis/internal.admin.acorn.io/v1/types.go [hunk 1]
```diff
@@ -15,6 +15,27 @@ func (in *ClusterComputeClassInstance) NamespaceScoped() bool {
...
+	}
+	in.SupportedRegions = append(in.SupportedRegions, region)
+	return true
+}
+
+// ForOtherRegions returns true if there are other regions that this instance is supported in.
+func (in *ClusterComputeClassInstance) ForOtherRegions(region string) bool {
+	for i, r := range in.SupportedRegions {
+		if r == region {
+			in.SupportedRegions = append(in.SupportedRegions[:i], in.SupportedRegions[i+1:]...)
```
Comment (g-linville) [inline pkg/apis/internal.admin.acorn.io/v1/types.go:32]: It looks like this is removing the matching region from the list. I'm a bit confused why a call to this function would remove a supported region from a ClusterComputeClassInstance.

File: pkg/server/registry/apigroups/acorn/volumes/class/strategy.go [hunk 2]
```diff
@@ -66,7 +65,6 @@ func (s *Strategy) list(ctx context.Context, namespace string, opts storage.List
 			projectDefaultExists = true
 		}
 		// Reset TypeMeta so proper GVK is reported
-		vc.TypeMeta = metav1.TypeMeta{}
```
Comment (g-linville) [inline pkg/server/registry/apigroups/acorn/volumes/class/strategy.go:69]: This comment should probably also be removed.

File: pkg/server/registry/apigroups/acorn/volumes/class/strategy.go [hunk 3]
```diff
@@ -80,7 +78,6 @@ func (s *Strategy) list(ctx context.Context, namespace string, opts storage.List
 			cvc.Default = false
 		}
 		// Reset TypeMeta so proper GVK is reported
-		cvc.TypeMeta = metav1.TypeMeta{}
```
Comment (g-linville) [inline pkg/server/registry/apigroups/acorn/volumes/class/strategy.go:83]: Also here

**Review by g-linville — APPROVED** [review]

**Comment by g-linville** [comment]
LGTM. Just had a couple nitpicks about comments that were left behind.

File: pkg/apis/api.acorn.io/v1/region.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   RegionSpec   `json:"spec,omitempty"`
+	Status RegionStatus `json:"status,omitempty"`
+}
+
+type RegionSpec struct {
+	Description string `json:"description,omitempty"`
+	AccountName string `json:"accountName,omitempty"`
```
Comment (StrongMonkey) [inline pkg/apis/api.acorn.io/v1/region.go:22]: Don't have the full context, Just want to call out that accountName is hub specific. Are we ok to make it into open source types?

File: pkg/apis/api.acorn.io/v1/region.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	RegionName  string `json:"regionName,omitempty"`
```
[the remark this answers is not in the document]
Comment (thedadams) [reply pkg/apis/api.acorn.io/v1/region.go:23]: `spec.regionName` is the actual region `us-east-2`, etc.

`region.Name` is the name of the region and will have something like `aws-` appended to it for future sake.

File: pkg/server/registry/apigroups/acorn/regions/strategy.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+func (s *strategy) Get(_ context.Context, _, name string) (types.Object, error) {
+	if name != "local" {
+		return nil, apierrors.NewNotFound(schema.GroupResource{
+			Group:    apiv1.SchemeGroupVersion.Group,
+			Resource: "regions",
+		}, name)
+	}
+
+	return &apiv1.Region{
```
Comment (StrongMonkey) [inline pkg/server/registry/apigroups/acorn/regions/strategy.go:27]: This is cool. I assume we don't persist region, we just return a local region.

Comment (thedadams) [reply pkg/server/registry/apigroups/acorn/regions/strategy.go:27]: Yep, that's right!

**Review by StrongMonkey — APPROVED** [review]
Looking good besides a question on the api

File: pkg/apis/api.acorn.io/v1/region.go (hunk 4 above)
Comment (thedadams) [reply pkg/apis/api.acorn.io/v1/region.go]: Fixed.

**Review by StrongMonkey — APPROVED** [review]

**Review by tylerslaton — APPROVED** [review]
Great work! Nothing stuck out to me as I reviewed outside of what was already mentioned (and addressed)

**Pull request merged**